src/test/java/org/czekalski/petapiintegration/config/AnimalPageCaseFactoryTest.java

```diff
@@ -116,5 +118,9 @@ void AddLinksAfterAllPages() {
                         CITY_ID);
         assertThat(animalResourcesListDtoEntityModel.getLink("prev").get().getHref())
                 .isEqualTo("/dogs/MA/Andover?size=1&page=128");
+
+        assertThrows(
```

Comment:
Test nie przechodzi, bo w 104 linijce pierwszym parametrem `PageImpl` jest niepusta lista. Gdy offset jest większy od total pages, to tutaj powinna być pusta lista. Bo jeżeli coś jest więcej, to znaczy, że jest więcej niż total. PageImpl to ogarnia i podbija Ci total do 130 stron :)

Reply:
Ja doszedłem do tego poprzez debugowanie. Najpierw sprawdziłem, która implementacja listy jest wykorzystana. Okazało się, że middle. A dlatego, że total było 130, a nie 128 jak podałeś. Potem sprawdziłem, w którym miejscu jest zamiana z 128 na 130. Okazało się, że w tym fragmencie kodu.

```
 public PageImpl(List<T> content, Pageable pageable, long total) {
        super(content, pageable);
        this.total = (Long)pageable.toOptional().filter((it) -> {
            return !content.isEmpty();
        }).filter((it) -> {
            return it.getOffset() + (long)it.getPageSize() > total;
        }).map((it) -> {
            return it.getOffset() + (long)content.size();
        }).orElse(total);
    }
```

W debugerze odpaliłem sobie każdy z tych warunków filtrowania i każdy z nich zwrócił `true`, więc pod uwagę nie była podana przez Ciebie wartość, ale wyliczona wartość `return it.getOffset() + (long)content.size();`, co daje 130

Reply:
Racja! poprawki wrzucę już na branch apiConsumer aby łatwiej było potem mergeować.

Reply:
Poprawiono:f555a44a0eb97d286f724e63260f0d0a7f9644dd branch: apiConsumer